node/pkg/query/response.go

```diff
@@ -847,3 +895,150 @@ func (left *EthCallWithFinalityQueryResponse) Equal(right *EthCallWithFinalityQu
+
+	buf := new(bytes.Buffer)
+	vaa.MustWrite(buf, binary.BigEndian, sar.SlotNumber)
+	vaa.MustWrite(buf, binary.BigEndian, sar.BlockTime.UnixMicro())
+	buf.Write(sar.BlockHash[:])
+
+	vaa.MustWrite(buf, binary.BigEndian, uint8(len(sar.Results)))
+	for _, res := range sar.Results {
+		vaa.MustWrite(buf, binary.BigEndian, res.Lamports)
+		vaa.MustWrite(buf, binary.BigEndian, res.RentEpoch)
+		vaa.MustWrite(buf, binary.BigEndian, res.Executable)
+		buf.Write(res.Owner[:])
+
+		vaa.MustWrite(buf, binary.BigEndian, uint32(len(res.Data)))
+		buf.Write(res.Data)
+	}
+
+	return buf.Bytes(), nil
+}
+
+// Unmarshal deserializes a Solana sol_account response from a byte array
+func (sar *SolanaAccountQueryResponse) Unmarshal(data []byte) error {
+	reader := bytes.NewReader(data[:])
+	return sar.UnmarshalFromReader(reader)
+}
+
+// UnmarshalFromReader  deserializes a Solana sol_account response from a byte array
+func (sar *SolanaAccountQueryResponse) UnmarshalFromReader(reader *bytes.Reader) error {
+	if err := binary.Read(reader, binary.BigEndian, &sar.SlotNumber); err != nil {
+		return fmt.Errorf("failed to read slot number: %w", err)
+	}
+
+	blockTime := int64(0)
+	if err := binary.Read(reader, binary.BigEndian, &blockTime); err != nil {
+		return fmt.Errorf("failed to read block time: %w", err)
+	}
+	sar.BlockTime = time.UnixMicro(blockTime)
+	if n, err := reader.Read(sar.BlockHash[:]); err != nil || n != SolanaPublicKeyLength {
+		return fmt.Errorf("failed to read block hash [%d]: %w", n, err)
+	}
+
+	numResults := uint8(0)
+	if err := binary.Read(reader, binary.BigEndian, &numResults); err != nil {
+		return fmt.Errorf("failed to read number of results: %w", err)
+	}
+
+	for count := 0; count < int(numResults); count++ {
+		var result SolanaAccountResult
+
+		if err := binary.Read(reader, binary.BigEndian, &result.Lamports); err != nil {
+			return fmt.Errorf("failed to read lamports: %w", err)
+		}
+
+		if err := binary.Read(reader, binary.BigEndian, &result.RentEpoch); err != nil {
+			return fmt.Errorf("failed to read rent epoch: %w", err)
+		}
+
+		if err := binary.Read(reader, binary.BigEndian, &result.Executable); err != nil {
+			return fmt.Errorf("failed to read executable flag: %w", err)
+		}
+
+		if n, err := reader.Read(result.Owner[:]); err != nil || n != SolanaPublicKeyLength {
+			return fmt.Errorf("failed to read owner [%d]: %w", n, err)
+		}
+
+		len := uint32(0)
+		if err := binary.Read(reader, binary.BigEndian, &len); err != nil {
+			return fmt.Errorf("failed to read data len: %w", err)
+		}
+		result.Data = make([]byte, len)
+		if n, err := reader.Read(result.Data[:]); err != nil || n != int(len) {
+			return fmt.Errorf("failed to read data [%d]: %w", n, err)
+		}
+
+		sar.Results = append(sar.Results, result)
+	}
+
+	return nil
+}
+
+// Validate does basic validation on a Solana sol_account response.
+func (sar *SolanaAccountQueryResponse) Validate() error {
+	// Not checking for SlotNumber == 0, because maybe that could happen??
```

Comment:
We've definitely seen slot zero on a few chains, but am unsure on solana. The guardiand rpc node would need to be hooked up to bigtable (which is very unlikely) as only a few guardians have access to it.

I think it is safe to verify != SlotNumber 0 on solana.

Reply:
Would like to leave as is.